components/camel-jdbc/src/main/java/org/apache/camel/component/jdbc/JdbcProducer.java

```diff
@@ -194,21 +193,12 @@ private boolean doCreateAndExecuteSqlStatementWithHeaders(Exchange exchange, Str
 
     private boolean doCreateAndExecuteSqlStatement(Exchange exchange, String sql, Connection conn) throws Exception {
 
+        Statement stmt = null;
         ResultSet rs = null;
         boolean shouldCloseResources = true;
 
         try {
-            // We might need to leave it open to allow post-processing of the result set. This is why we
```

Comment:
This is a bit wrong, in case you use JdbcOutputType.StreamList then we need to close it in the UoW, eg its when setResultSet returns false.

But I think we can keep the closing in the finally block if setResultSet returns true. Then its closed eager in finally, and then again in UoW done, but its already closed and isClosed will return true.

Reply:
if `setResultSet` resturns `false`, it has added `exchange.getExchangeExtension().addOnCompletion(new ResultSetIteratorCompletion(iterator))`. So I think we don't need to close it in the UoW here.

Reply:
Or it should be better to move these codes
```java
try {
...
shouldCloseResources = setResultSet(exchange, conn, rs);
...

} finally {
            if (shouldCloseResources) {
                closeQuietly(rs);
                closeQuietly(stmt);
            } else {
                exchange.getExchangeExtension().addOnCompletion(new SynchronizationAdapter() {
                @Override
                public void onDone(Exchange exchange) {
                    closeQuietly(rs);
                    closeQuietly(stmt);
                }
            });
           }
        }       

```

Reply:
Yeah then the closing logic is at the same place and easier to ensure is correct